src/python/pants/base/exception_sink.py

```diff
@@ -371,8 +371,10 @@ def trapped_signals(cls, new_signal_handler):
 
     NB: This method calls signal.signal(), which will crash if not called from the main thread!
     """
+    # NB: It's possible for the `.reset_signal_handler()` call to raise, but it's not yet clear when
```

Comment:
The comment is sort of besides the point. The code is actually now in a normal pattern that requires no further explanation.

Reply:
That makes sense now -- thank you!